torch/testing/_internal/optests/aot_autograd.py

```diff
@@ -118,7 +118,8 @@ def check(args, ignore_failure=False):
             raise
 
         # See https://github.com/pytorch/pytorch/pull/98960#issuecomment-1505962215
-        if all(x is None for x in orig_grad):
+        any_non_leaves = any(x.requires_grad and not x.is_leaf for x in args)
+        if all(x is None for x in orig_grad) and any_non_leaves:
```

Comment:
cc @albanD, I'm hoping what I put in the PR description makes sense to you :) (after re-reading some of the previous commentary)